react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+
+export default function ValidatedInput() {
+  const [isTooShort, setIsTooShort] = useState(false);
+  const [hasNoPassword, setHasNoPassword] = useState(false);
```

Comment:
It would be better to have just one state , which is the password, and then calculate the errors at render time.